File: week3/atakantekatan/206. Reverse Linked List.py
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        self.rev = None
        while head:
            new = ListNode(head.val)
            if self.rev is None:
                self.rev = new
            else:
                new.next = self.rev
                self.rev = new
            head = head.next
        return self.rev

File: week3/Can-Yavuz/206-reverse-linked-list.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode ReverseList(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }
        ListNode prev = null;
        ListNode current = head;
        
         while (current != null) {
            ListNode nextNode = current.next;
            current.next = prev;
            prev = current;
            current = nextNode;
        }
        
        return prev;
    }
}